File: practica4/CODIGO/PYTHON/histogramas/eq_hist_cor.py
import cv2
import numpy as np

# Lemos a imaxe
filename = "../../data/images/dark-flowers.jpg"
im = cv2.imread(filename)
# Comprobamos se se leu a imaxe
if im is None:
    print("Non poiden ler a imaxe")

imEq = np.zeros_like(im)

# Realizamos ecualizacion separadamente das canles
for i in range(0,3):
    imEq[:,:,i] = cv2.equalizeHist(im[:,:,i])

cv2.imshow("Imaxe Orixinal",im)
cv2.waitKey(0)
cv2.imshow("Histograma ecualizada",imEq)
cv2.waitKey(0)

# Lemos a imaxe de cor
filename = "../../data/images/dark-flowers.jpg"
im = cv2.imread(filename)
# Comprobamos se se leu a imaxe
if im is None:
    print("Non poiden ler a imaxe")

# Convertemos HSV
imhsv = cv2.cvtColor(im, cv2.COLOR_BGR2HSV)

# Ecualozamos so a candle V
imhsv[:,:,2] = cv2.equalizeHist(imhsv[:,:,2])

# Convertemos o formato BGR
imEq = cv2.cvtColor(imhsv, cv2.COLOR_HSV2BGR)

cv2.imshow("Imaxe Orixinal",im)
cv2.waitKey(0)
cv2.imshow("Histograma ecualizada",imEq)
cv2.waitKey(0)

File: codigos_clase/practica4/CODIGO/C++/eq_hist/eq_hist_cor.cpp
#include <iostream>
#include <opencv2/opencv.hpp>

using namespace std;
using namespace cv;

int main(){
string filename = "../../../data/images/dark-flowers.jpg";
Mat img = imread(filename);
Mat imEq = img.clone();

// Realizamos a ecualizacion para as canles por separado
vector<Mat> imgChannels(3);
vector<Mat> imEqChannels(3);

split(img,imgChannels);
split(imEq,imEqChannels);

for (int i=0; i<3; i++){
    equalizeHist(imgChannels[i],imEqChannels[i]);
}

merge(imgChannels,img);
merge(imEqChannels,imEq);

imshow("Imaxe Orixinal",img);
waitKey(0);
imshow("Histograma ecualizado", imEq);
waitKey(0);

filename = "../../../data/images/dark-flowers.jpg";
img = imread(filename);
Mat imhsv;

cvtColor(img, imhsv, COLOR_BGR2HSV);

vector<Mat> imhsvChannels(3);
split(imhsv,imhsvChannels);

// Ecualizacion so da candle V 
equalizeHist(imhsvChannels[2],imhsvChannels[2]);
merge(imhsvChannels,imhsv);
// Convertimos ao formato BGR
cvtColor(imhsv,imEq, COLOR_HSV2BGR);

imshow("Imaxe Orixianal",img);
waitKey(0);
imshow("Histograma ecualizado",imEq);
waitKey(0);

return 0;
}
